src/executor/executor.cc

```diff
@@ -0,0 +1,66 @@
+#include "executor/executor.hh"
+
+namespace sim {
+
+static void executeAdd(const Instruction &inst, State &state) {
+  RegVal rs1 = state.regs.get(inst.rs1);
+  RegVal rs2 = state.regs.get(inst.rs2);
+  state.regs.set(inst.rd, rs1 + rs2);
+}
+
+static void executeSub(const Instruction &inst, State &state) {
+  RegVal rs1 = state.regs.get(inst.rs1);
+  RegVal rs2 = state.regs.get(inst.rs2);
+  state.regs.set(inst.rd, rs1 - rs2);
+}
+
+static void executeMul(const Instruction &inst, State &state) {
+  RegVal rs1 = state.regs.get(inst.rs1);
+  RegVal rs2 = state.regs.get(inst.rs2);
+  state.regs.set(inst.rd, rs1 * rs2);
+}
+
+static void executeDiv(const Instruction &inst, State &state) {
+  RegVal rs1 = state.regs.get(inst.rs1);
+  RegVal rs2 = state.regs.get(inst.rs2);
+  state.regs.set(inst.rd, rs1 / rs2);
+}
+
+static void executeLw(const Instruction &inst, State &state) {
+  RegVal rs1 = state.regs.get(inst.rs1);
+  RegVal offset = inst.imm;
+  state.regs.set(inst.rd, state.mem.load(rs1 + offset));
```

Comment:
Immediate should be cast to signed type
@Bryanskiy, should I provide a method `Instruction::getSignedImm()` or you will cast it on `Executor`'s side?

Reply:
It can't be implemented in `Executor` or `Instruction` because `using RegVal = std::uin32_t` loses information about sign bit. We should provide something like `bool signBit` or 'using immVal = std::int32_t' in `Instruction` and obtain sign bit by decoder.

Reply:
I suggest third variant
common.hh:
```c++
template <std::unsigned_integral T>
constexpr auto signCast(T val) 
{
  return static_cast<std::make_signed_t<T>>(val);
}
```

Reply:
done